lume_model/models/gp_model.py

```diff
@@ -0,0 +1,412 @@
+    passed to this class. For ModelListGP, the passed input_transformers and output_transformers will be applied to all
+    models in the list. If different transformers are needed for different models, the models should be instantiated
+    separately using Botorch's input_transform and outcome_transform attributes before creating the model list.
+
+    Args:
+        model: A single task GPyTorch model or BoTorch model.
+        input_transformers: List of input transformers to apply to the input data. Optional, default is None.
+        output_transformers: List of output transformers to apply to the output data. Optional, default is None.
+    """
+
+    model: SingleTaskGP | MultiTaskGP | ModelListGP
+    input_transformers: list[ReversibleInputTransform | torch.nn.Linear] | None = None
+    output_transformers: (
+        list[OutcomeTransform | ReversibleInputTransform | torch.nn.Linear] | None
+    ) = None
+    check_transforms: bool = True
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.input_transformers = (
+            [] if self.input_transformers is None else self.input_transformers
+        )
+        self.output_transformers = (
+            [] if self.output_transformers is None else self.output_transformers
+        )
+        if self.check_transforms:
+            self._check_transforms()
+
+    @field_validator("model", mode="before")
+    def validate_gp_model(cls, v):
+        if isinstance(v, (str, os.PathLike)):
+            if os.path.exists(v):
+                v = torch.load(v, weights_only=False)
+            else:
+                raise OSError(f"File {v} is not found.")
+        return v
+
+    @field_validator("input_transformers", "output_transformers", mode="before")
+    def validate_transformers(cls, v):
+        if not isinstance(v, list):
+            raise ValueError("Transformers must be passed as list.")
+        loaded_transformers = []
+        for t in v:
+            if isinstance(t, (str, os.PathLike)):
+                if os.path.exists(t):
+                    t = torch.load(t, weights_only=False)
+                else:
+                    raise OSError(f"File {t} is not found.")
+            loaded_transformers.append(t)
+        v = loaded_transformers
+        return v
+
+    def _check_transforms(self):
+        """
+        Check the input and output transforms for the model.
+
+        If the trained model already has transform attributes,
+        a warning will be printed to alert the user if transforms are also passed.
+
+        This will only display a warning if the transforms are specified twice. It is up to the user
+        to adjust the specified transforms if necessary and reinstantiate.
+        """
+        # SingleTask and MultiTask
+        if (
+            hasattr(self.model, "input_transform")
+            and self.model.input_transform is not None
+        ):
+            if self.input_transformers is not None:
```

Comment:
can we specify in the docs that the input/output transforms are applied outside of the underlying model (regardless of what is inside the model), it would make these warning messages unnecessary. I think its straightforward to understand that submodules of the model class will be used inside that class